modules/network/src/main/java/org/apache/ignite/internal/network/netty/ConnectionManager.java

```diff
@@ -486,4 +497,22 @@ public Map<ConnectorKey<String>, NettySender> channels() {
     public void initiateStopping() {
         stopping.set(true);
     }
+
+    /**
+     * Closes physical connections with an Ignite node identified by the given ID (it's not consistentId,
+     * it's ID that gets regenerated at each node restart).
+     *
+     * @param id ID of the node.
+     */
+    public void closeConnectionsWith(String id) {
+        List<Entry<ConnectorKey<String>, NettySender>> entriesToRemove = channels.entrySet().stream()
```

Comment:
Can we use `removeIf` instead?

Reply:
This would mean the following:

1. The predicate would have a side effect (which is weird; `close()` is probably idempotent, but still)
2. `close()` invoked inside a concurrent map, so probably under a lock. This is scary to do; also, channel closure might trigger some events (reacting to the closure); if they go to the same map, a deadlock might happen

I don't think that the convenience of `removeIf()` is worth it given the necessity to reason about all this.